bionic/executor.py

```diff
@@ -0,0 +1,88 @@
+"""
+Contains Executor class, that handles all the objects related to parallel
+processing.
+"""
+
+import logging
+
+from .logging import LoggingReceiver, WorkerProcessLogHandler
+from .optdep import import_optional_dependency
+from .util import oneline
+
+_executor = None
+
+
+def get_reusable_executor(process_manager):
+    global _executor
+
+    # TODO: For now, executor is always the same. But once the worker
+    # count is configurable, we should try and update the loky executor
+    # and log a warn if the eecutor changed.
+    if _executor is None:
+        _executor = Executor(process_manager)
+
+    return _executor
+
+
+class Executor:
+    """
+    Encapsulates all objects related to parallel processing sans manager
+    in one place.
+    """
+
+    # TODO move process manager into the executor class.
```

Comment:
What's stopping us from doing this now? Are we waiting until we can get rid of the `SynchronizedSet` usage in `TaskKeyLogger`?

Reply:
No, I just did this in a follow up task. I can merge that in this PR if you'd like.

Reply:
Oh, got it. I'm happy to look at that one separately.